e2e/cypress/plugins/smtpTester.js

```diff
@@ -29,6 +33,86 @@ const initialize = (on, config) => {
       return lastEmail[email] || null;
     },
   });
+
+  on('task', {
+    async compareImages({ expectedImageFilepath, newImageContent, updateExpectedImage = false }) {
+      const inputBuffer = Buffer.from(newImageContent.data);
+      if (updateExpectedImage) {
+        fs.writeFileSync(expectedImageFilepath, inputBuffer);
+        return true;
+      }
+
+      const inputImage = await Jimp.read(inputBuffer);
+      const expectedImage = await Jimp.read(expectedImageFilepath);
+
+      const pixelDiff = diff(expectedImage, inputImage, 0.3);
+      return pixelDiff.percent <= 0.1;
+    },
+  });
+
+  on('task', {
+    async compareCSVs({ expectedCSVFilepath, newCSVContent, updateExpectedCSV = false }) {
+      const inputBuffer = Buffer.from(newCSVContent.data);
+      if (updateExpectedCSV) {
+        await fs.writeFileSync(expectedCSVFilepath, inputBuffer);
+        return true;
+      }
+
+      const inputCSV = toCSV(inputBuffer);
+      const expectedCSV = toCSV(fs.readFileSync(expectedCSVFilepath));
+
+      if (inputCSV.length !== expectedCSV.length) {
+        return false;
+      }
+
+      for (let i = 0; i < expectedCSV.length; i++) {
+        const line = expectedCSV[i];
+        for (let j = 0; j < line.length; j++) {
+          if (line[j] !== inputCSV[i][j]) {
+            return false;
+          }
+        }
+      }
+
+      return true;
+    },
+  });
+
+  on('task', {
+    async comparePDFs({ expectedPDFFilepath, newPDFContent, updateExpectedPDF = false }) {
+      const inputBuffer = Buffer.from(newPDFContent.data);
+      if (updateExpectedPDF) {
+        fs.writeFileSync(expectedPDFFilepath, inputBuffer);
+        return true;
+      }
+
+      const inputDoc = await pdf(inputBuffer);
+      const expectedDoc = await pdf(expectedPDFFilepath);
+
+      cleanUpPDFText(inputDoc);
```

Comment:
I saw the comment below but can we also add a comment here about why this is needed? Or maybe change the function name to something more meaningful like "removeHeaderText"